src/System.Private.CoreLib/shared/System/Threading/Tasks/Sources/ManualResetValueTaskSourceCore.cs

```diff
@@ -106,7 +107,7 @@ public TResult GetResult(short token)
         /// <param name="state">The state object to pass to <paramref name="continuation"/> when it's invoked.</param>
```

Comment:
can `GetResult` return null? It returns _result and we're annotating it as `MaybeNull`. Should we add a `MaybeNull` to its return type?

Reply:
Ah, we shouldn't as that will be stated by the generic type parameter used to create the instance.

Reply:
Right.  We should only end up returning _result if the instance has been correctly completed, in which case _result should only still be null if TResult is instantiated as nullable.